cai_causal_graph/time_series_causal_graph.py

```diff
@@ -0,0 +1,888 @@
+        # if the source and destination time series nodes do not exist, create them
+        if source is not None and not self.node_exists(source):
+            source_meta = None
+            if isinstance(source, HasMetadata):
+                source_meta = source.get_metadata()
+                source = self.add_node(source, meta=source_meta)
+
+        if destination is not None and not self.node_exists(destination):
+            destination_meta = None
+            if isinstance(destination, HasMetadata):
+                destination_meta = destination.get_metadata()
+                destination = self.add_node(destination, meta=destination_meta)
+
+        edge = super().add_edge(source, destination, edge_type=edge_type, meta=meta, edge=edge, **kwargs)
+
+        return edge
+
+    @classmethod
+    def from_causal_graph(cls, causal_graph: CausalGraph) -> TimeSeriesCausalGraph:
+        """
+        Return a time series causal graph from a causal graph.
+
+        This is useful for converting a causal graph from a single time step into a time series causal graph.
+
+        :param causal_graph: The causal graph.
+        """
+
+        sepsets = deepcopy(causal_graph._sepsets)
+
+        # copy nodes and make them TimeSeriesNodes
+        ts_cg = cls()
+        for node in causal_graph.get_nodes():
+            meta = deepcopy(node.meta)
+            # get the variable name and lag from the node name
+            variable_name, lag = get_variable_name_and_lag(node.identifier)
+            node = TimeSeriesNode(variable_name=variable_name, time_lag=lag, meta=meta)
+            ts_cg.add_node(node)
+
+        # copy edges
+        for edge in causal_graph.get_edges():
+            source = ts_cg.get_node(edge.source)
+            destination = ts_cg.get_node(edge.destination)
+            ts_cg.add_edge(source, destination, meta=edge.meta)
+
+        ts_cg._sepsets = sepsets
+
+        return ts_cg
+
+    @staticmethod
+    def from_adjacency_matrix(
+        adjacency: numpy.ndarray,
+        node_names: Optional[List[Union[NodeLike, int]]] = None,
+    ) -> TimeSeriesCausalGraph:
+        """
+        Construct a `cai_causal_graph.time_series_causal_graph.TimeSeriesCausalGraph` instance from an adjacency matrix
+        and optionally a list of node names.
+
+        :param adjacency: A square binary numpy adjacency array.
+        :param node_names: A list of strings, `cai_causal_graph.interfaces.HasIdentifier`, and/or integers which can be
+            coerced to `cai_causal_graph.graph_components.Node`.
+        :return: A `cai_causal_graph.time_series_causal_graph.TimeSeriesCausalGraph` object.
+        """
+        graph = CausalGraph.from_adjacency_matrix(adjacency, node_names=node_names)
+        return TimeSeriesCausalGraph.from_causal_graph(graph)
+
+    @staticmethod
+    def from_adjacency_matrices(
+        adjacency_matrices: Dict[int, numpy.ndarray],
+        variable_names: Optional[List[Union[NodeLike, int]]] = None,
```

Comment:
variable_names should be called node_names for consistency with the previous method, but I am not sure, as here we have several lags for the same variable so several nodes potentially, so I can see the rationale of using variable_names.

Reply:
Marking this as solved as variables differ from nodes in that they do not contain the lag, so I can see why the parameter name was chosen.